docs/DEVELOPMENT.md

```diff
@@ -1,46 +1,38 @@
 # Development
 
-This doc explains how to set up a development environment so you can get started.
+This doc explains how to set up a development environment for NumaFlow.
 
-## Prerequisites
-
-Follow the instructions below to set up your development environment.
-
-### Install requirements
-
-These tools are required for development.
+### Install required tools
 
 1. [`go`](https://golang.org/doc/install): 1.17+.
 1. [`git`](https://help.github.com/articles/set-up-git/): For source control.
 1. [`kubectl`](https://kubernetes.io/docs/tasks/tools/install-kubectl/): For managing development environments.
 1. [`protoc`](https://github.com/protocolbuffers/protobuf): For compiling protocol buffers.
+1. [`k3d`](https://k3d.io/) for local development, if needed
 
-### Create a dev cluster
-
-We recommend to use [`k3d`](https://k3d.io/) to create a cluster for development.
-
-After installing `k3d`, use following command to create a kubernetes cluster with default name `k3s-default`.
+### Create a k8s cluster with k3d if needed
 
 ```shell
+# Create a cluster with default name k3s-default
 k3d cluster create -i rancher/k3s:v1.21.7-k3s1
 
-# kubeconfig
+# Get kubeconfig for the cluster
 k3d kubeconfig get k3s-default
 ```
 
-## Useful Commands
+### Useful Commands
 
 - `make build`
-  Build binaries into `./dist` directory.
+  Binaries are placed in `./dist`.
 
 - `make codegen`
-  If there's any model changes (`./pkg/api/`), this command need to be run to regenerate code.
+  Run after making changes to `./pkg/api/` Can we add a Makefile rule to do this automatically?
```

Comment:
There's a CI job to check if this is done or not.

Reply:
What about for users running it manually on their laptop though?

Reply:
`codegen` is already a rule, developers need to run it manually, do you mean adding a rule to check if `make codegen` is run?

Reply:
The comment seems to say that if anything in pkg/api changes, users need to make codegen before they do make build? Can't make build automatically run codegen if anything changes?

Reply:
Do not want to make `make build` run codegen automatically because:

1. `codegen` does not create any new files, but regenerate the content of several files, so we can not use makefile rule prerequisites to check file existence, but have to run it every time if we do that.
2. It takes too long to run codegen (around 1 min), and there's no need to run it every time - only needed when changes in pkg/api happen.

Reply:
If it always regenerates a particular file, we can use a makefile rule for that file.
Otherwise, we could touch a dummy file each time codegen runs and create a rule for the dummy file.
Definitely don't want to run codegen every time.